dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AppHubController.java

```diff
@@ -80,9 +82,12 @@
         return appHubService.getAppHub();
     }
 
-    @GetMapping( value = "/{apiVersion}", produces = "application/json" )
-    public @ResponseBody String getAppHubApiResponse( @PathVariable String apiVersion, @RequestParam String query )
+    @GetMapping( value = "/{apiVersion}/**", produces = "application/json" )
```

Comment:
This seems cleaner to me, thanks!  I'll leave it to @larshelge and @netroms to decide if this has negative implications for security, api cleanliness, or performance - if so we can use the `?query=` version so long as additional query parameters are transparently forwarded to the `apps.dhis2.org` API.

Reply:
(I don't think we can use regex here, so `/appHub/v{apiVersion}/**` perhaps?

Reply:
Right, I think `v{apiVersion}` might not add a lot value if you can still substitute `apiVersion` with anything.

If this is important we can do validation in application logic. Is it safe to assume `v\d+` for all future?

Reply:
Added API version validation now in service layer (`v\d+`).